src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -162,7 +162,29 @@ private function handleRaw(Request $request, int $type = self::MASTER_REQUEST)
                 if (null === $response) {
                     $msg .= ' Did you forget to add a return statement somewhere in your controller?';
                 }
-                throw new \LogicException($msg);
+
+                $e = new \LogicException($msg);
+
+                if (!$controllerDefinition = $this->parseControllerDefinition($controller)) {
+                    throw $e;
+                }
+
+                $r = new \ReflectionProperty(\Exception::class, 'trace');
+                $r->setAccessible(true);
+                $r->setValue($e, array_merge(array(
+                    array(
+                        'line' => $e->getLine(),
```

Comment:
This could be even more accurate, passing the line of the call_user_fun above (using `__LINE__` next to it.)
Aren't they other keys we should put in the frame? How does it look when thrown from inside the controller?

Reply:
fixed